source/simulator/solver_schemes.cc

```diff
@@ -337,6 +337,60 @@ namespace aspect
     if (fields_advected_by_particles.size() > 0)
       interpolate_particle_properties(fields_advected_by_particles);
 
+    // If elasticity is switched on, the stress tensor can have components
+    // with very large values and components that are and stay practically zero.
+    // Therefore, we include all fields that represent stress tensor components
+    // of the same stress tensor in the computation of the initial residual
+    // for the fields that belong to that tensor. In other words, we compute an
+    // averaged initial residual using those fields that belong to the ve_stress tensor,
+    // and one for those belonging to the ve_stress_old tensor.
+    // TODO ve_stress_old values are not independent components of the solution vector,
+    // do we need to compute a residual for them at all?
```

Comment:
Okay, I'll now have a look whether we can get around the ve_stress_old fields by applying the rotation to the advected fields whenever we need them, otherwise I can set their residuals to 0.

Reply:
I've set their residuals to 0, as we decided to keep the ve_stress_old fields, so that we can extract the velocity gradients at their correct locations.